hbase-common/src/main/java/org/apache/hadoop/hbase/io/ByteBuffAllocator.java

```diff
@@ -171,6 +171,9 @@ public static ByteBuffAllocator create(Configuration conf, boolean reservoirEnab
       // that by the time a handler originated response is actually done writing to socket and so
       // released the BBs it used, the handler might have processed one more read req. On an avg 2x
       // we consider and consider that also for the max buffers to pool
+      if (poolBufSize == 0) {
+        throw new IllegalArgumentException("poolBufSize cannot be zero");
```

Comment:
The error message that mentions an internal variable name may be confusing. It may be better to refer to `BUFFER_SIZE_KEY` that the user can actually set through a configuration.

Reply:
Got it, prof.